sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -276,6 +278,29 @@ def transform(self, X):
         return sp.csr_matrix(X)
 
 
+class BinaryDecisionTreeClassifier(DecisionTreeClassifier):
+    def _more_tags(self):
+        return {'binary_only': True}
+
+    def fit(self, X, y, sample_weight=None):
+        super().fit(X, y, sample_weight)
+        X, y = check_X_y(X, y, y_numeric=False)
+        check_classification_targets(y)
+        self.classes_, y = np.unique(y, return_inverse=True)
+        n_trim_classes = np.count_nonzero(np.bincount(y, sample_weight))
+        if n_trim_classes != 2:
+            raise ValueError("y contains %d class after sample_weight "
+                             "trimmed classes with zero weights, while 2 "
+                             "classes are required." % n_trim_classes)
+        return self
+
+    def predict_proba(self, X):
```

Comment:
You should not need to redefine `predict_proba` or `predict`  since you're inheriting from `DecisionTreeClassifier`

Reply:
Yep, removed